### ISSUE TITLE
Previewing does not work occasionally

### ISSUE THREAD

**Issue by subev** [dyng/ctrlsf.vim#271]
![ctrlsf-preview-issue](https://user-images.githubusercontent.com/1330872/61951487-c4f96500-afb9-11e9-831b-07065f60c8c6.gif)

1. Update to the latest version of CtrlSF. <- Done
2. Update your backend(ag/ack/pt/rg) to their latest version. <- Done
3. Enable debug mode to try to find what is going wrong yourself. You can enable debug mode by `let g:ctrlsf_debug_mode = 1`. (I did but it is not related to the issue)

#### Previewing a match is SOMETIMES working.

The plugin is working perfectly to show the results, the issue is with the preview functionality.
On some of the matches in the results window when I press 'p', sometimes it throws an exception saying
```
Error detected while processing function ctrlsf#JumpTo[32]..<SNR>306_PreviewFile[12]..ctrlsf#win#MoveCursorCentral:
line    2:
E382: Cannot write, 'buftype' option is set 
```
I tried many things to see where the issue is coming from:
- use just the CtrlSF plugin, and comment out all the other from my vimrc
- do not use any of the ctrlsf variables and/or go with the default setup and different (ag) backend - same story

I searched the internet but cannot find anything related. I am willing to help as much as I can to improve the plugin.

#### Things about my system and environment

|  field  |                value                 |
|:-------:|:------------------------------------:|
|   os    |       *MacOS Mojave 10.14.5*        |
|   mvim   | *MacVim 8.1.1517* |
| backend |             *rg 11.0.1*             |
| locale  |            *en_US.UTF-8*             |

- log:

    ```vim
    Tokens: ['vim']
    ParsedResult: {'pattern': 'vim'}
    Options: {'pattern': 'vim', '_vimregex': '\V\cvim', '_vimhlregex': {'normal': '\V\c\(\^\d\+:\.\*\)\@<=vim', 'compact': '\V\c\(|\d\+ col \
    d\+|\.\*\)\@<=vim'}}
    ExecCommand: rg -C 3  --smart-case --fixed-strings --follow --no-heading --color never --line-number -H -- "vim" "/Users/petur/dotfiles"
    TimerStarted: id=12
    Searching...
    DefactorizeLine: [Factor]: [/Users/petur/dotfiles/README.md, 12, ], [Orig]: /Users/petur/dotfiles/README.md-12-  


    ```

- vimrc:

    ```vim
    Plug 'dyng/ctrlsf.vim'
    let g:ctrlsf_ackprg = 'rg'
    "let g:ctrlsf_debug_mode = 1
    let g:ctrlsf_mapping = {
      \ "next": "n",
      \ "prev": "N",
      \ "vsplit": "s",
      \ }
    let g:ctrlsf_auto_focus = {
        \ "at" : "start"
        \ }
    let g:ctrlsf_confirm_save = 0
    ```

- file:

    ```
    dotfiles
    
    ========
    
    unix settings 
    
    ln -s dotfiles/.bashrc ~/.bashrc
    
    ln -s dotfiles/.zshrc ~/.zshrc
    
    ln -s dotfiles/.bash_profile ~/.bash_profile
    
    ln -s dotfiles/.gitconfig ~/.gitconfig
    
    ln -s dotfiles/.vimrc ~/.vimrc
    
    ln -s dotfiles/.gvimrc ~/.gvimrc
    
    ln -s dotfiles/.inputrc ~/.inputrc
    
    ln -s dotfiles/.screenrc ~/.screenrc
    
    ln -s dotfiles/.agignore ~/.agignore
    
    ln -s dotfiles/.agignore ~/.rgignore


    ```

**Comment by subev** [comment]
Alright, I spend few more hours to really get to the bottom of this. It turns out that the issue is because I mapped my numbers to do what their SHIFT equivalent does (don't ask me why :))).
```vim
    nnoremap 2 :w<CR>
    nnoremap 3 #
    vnoremap 3 #
    nnoremap 4 $
    vnoremap 4 $
    nnoremap 6 ^
    vnoremap 6 ^
    nnoremap 8 *
    vnoremap 8 *
```
Why would it affect this plugin and not the others, I explicitly used **noremap** to avoid messing up the native commands?
Could you suggest a workaround?

**Comment by subev** [comment]
I even tried to [restore](https://vi.stackexchange.com/questions/18468/unmapping-the-global-mappings-on-a-per-buffer-basis) them for the __CtrlSF__ buffer but although the shortcuts no longer work, the error messages still appear when I try to preview the file. 
```vim
    au BufEnter __CtrlSF__ nnoremap <buffer> 2 2
    au BufEnter __CtrlSF__ nnoremap <buffer> 3 3
    au BufEnter __CtrlSF__ nnoremap <buffer> 4 4
    au BufEnter __CtrlSF__ nnoremap <buffer> 5 5
    au BufEnter __CtrlSF__ nnoremap <buffer> 6 6
    au BufEnter __CtrlSF__ nnoremap <buffer> 8 8
    au BufEnter __CtrlSF__ nnoremap <buffer> 9 9
    au BufEnter __CtrlSF__ nnoremap <buffer> 0 0
```

**Comment by subev** [comment]
Okey I found a workaround, it turned out the buffer that was causing issues was the preview buffer. So overriding the shortcuts for it makes it work:
```vim
    au BufEnter __CtrlSFPreview__ noremap <buffer> 2 2
    au BufEnter __CtrlSFPreview__ noremap <buffer> 3 3
    au BufEnter __CtrlSFPreview__ noremap <buffer> 4 4
    au BufEnter __CtrlSFPreview__ noremap <buffer> 5 5
    au BufEnter __CtrlSFPreview__ noremap <buffer> 6 6
    au BufEnter __CtrlSFPreview__ noremap <buffer> 8 8
    au BufEnter __CtrlSFPreview__ noremap <buffer> 9 9
    au BufEnter __CtrlSFPreview__ noremap <buffer> 0 0
```

**Comment by dyng** [comment]
Sorry for this late reply, I had been busy in packing and moving house last week, so have few time to approach this issue.

Thank you for your detailed issue description, and you have figured out the true reason: normal commands interfered with your maps, I have fixed this problem in #273 , please check out that branch and verify if the problem is solved.

**Comment by subev** [comment]
It works perfectly! Thank you!
